### ISSUE TITLE
FastBoot support

### ISSUE THREAD

**Issue by lolmaus** [fivetanley/ember-cli-dotenv#24]
Hi!

I'm trying to use `ember-cli-dotenv` with FastBoot.

My goal is to pass a private key to the FastBoot-driven app without letting the key appear in the public app's source.

With FastBoot, the `clientAllowedKeys` seems to be ignored: all vars from `.env` end up in the source even if they're not listed in `clientAllowedKeys`.

**Comment by fivetanley** [comment]
Hi @lolmaus. Do you have a reproduction project? Unfortunately I'm not using this with any FastBoot projects at the moment.

**Comment by SergeAstapov** [comment]
@kratiahuja do you mind to share some thoughts on this issue? I faced the same issue and I'd like to contribute to the addon to support that feature but have a difficult time thinking about how to integration this into ember-cli-fastboot

**Comment by kratiahuja** [comment]
How about write it to the package.json of Fastboot as part of the app config? I am working on exposing an API on fastboot where you can write additional config at build time and read it in Fastboot via FastBoot.config

**Comment by SergeAstapov** [comment]
> How about write it to the package.json of Fastboot as part of the app config?

That would work and that's what I;d like to achieve!
The main problem here is: how to make some config to appear in package.json and **do not appear** in `<meta name="{{modulePrefix}}/config/environment">`?
I see that package.jsom is populated with the same content as `<meta name="{{modulePrefix}}/config/environment">` at build time (according to [ember-cli-fastboot/index.js](https://github.com/ember-fastboot/ember-cli-fastboot/blob/494b8a14f8ae27a8c2caec7e75beb51da8a358cc/index.js#L248))

I'm looking into some option to allow an addon like this one to tell `ember-cli-fastboot` "Hey, here is some extra params you should include in package.json" but at the same time these vars should be ignored by ember-cli itself and not added to resulting `index.html`?
Or maybe the addon like this one can run on Broccoli tree **after** `ember-cli-fastboot` and update `package.json` after it's produced. Not sure what would be the best option.

> I am working on exposing an API on Fastboot where you can write additional config at build time and read it in Fastboot via FastBoot.config

@kratiahuja are you talking about https://github.com/ember-fastboot/ember-cli-fastboot/pull/515? That also could be a workaround. If I got it correctly I would need to create `config/environment` module with some prefix, like `{{app-name}}-fastboot/config/environment`.
Then there would be similar problem - how to make this `{{app-name}}-fastboot/config/environment` not available in client-side app and not delivered to browser at all?

I can do first option with "patching" `package.json` at build time but need some guidance.

**Comment by kratiahuja** [comment]
> are you talking about ember-fastboot/ember-cli-fastboot#515

Yes. You don't need to create config/environment module. you can just write key value pair in the config.

**Comment by kratiahuja** [comment]
@SergeAstapov it might be best to discuss this over slack. My point is `package.json` generated is never shipped to the browser so you could always write the contents in there and expose it in the server side app. Example is `hostWhitelists` in package.json as well.

**Comment by SergeAstapov** [comment]
@kratiahuja seems that I was confused by example in `README.md` from https://github.com/ember-fastboot/ember-cli-fastboot/pull/515 PR:
> Addons wishing to use this hook simply need to return a unique identifier for the configuration with the configuration.
>
> ```js
> fastbootConfigTree() {
>  return {
>    '<engine-name>/config/environment': {
>      'foo': 'bar'
>    }
>  }
>}
>```

and did not pay attention to the fact that `'<engine-name>/config/environment'` would be a key on `appConfig` and available in the app in FastBoot mode via
```
import ENV from 'app-name/config/environment';
...
ENV['<engine-name>/config/environment']
```

Now I've the idea and would proceed with PR to `ember-cli-dotenv` with `fastbootConfigTree` hook support.

**Comment by kratiahuja** [comment]
@SergeAstapov `fastbootConfigTree` hook hasn't landed yet. You would need to wait for it to land and then use it with checking the minimal version of ember-cli-fastboot that supports the hook in the host app.

**Comment by SergeAstapov** [comment]
@kratiahuja thanks for note about checking the minimal version of ember-cli-fastboot!
We'll need to mention in PR that it can land only after https://github.com/ember-fastboot/ember-cli-fastboot/pull/515 is merged and new releases of fastboot and ember-cli-fastboot are out.

Similar way how you cross linked PR's between fastboot and ember-cli-fastboot.

**Comment by SergeAstapov** [comment]
Closed via #50. Published 2.2.0
